src/output-json-alert.c

```diff
@@ -287,27 +285,30 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
                 }
 
                 if (json_output_ctx->flags & LOG_JSON_PAYLOAD) {
+                    uint8_t printable_buf[payload->offset + 1];
+                    uint32_t offset = 0;
+                    PrintStringsToBuffer(printable_buf, &offset,
+                                     sizeof(printable_buf),
+                                     (unsigned char *)payload, payload->offset);
                     json_object_set_new(js, "payload_printable",
-                                        json_string((char *)payload->buffer));
+                                        json_string((char *)printable_buf));
                 }
             } else {
                 /* This is a single packet and not a stream */
-                unsigned char packet_buf[p->payload_len + 1];
-                uint32_t offset = 0;
-
-                PrintStringsToBuffer(packet_buf, &offset,
-                                     p->payload_len + 1,
-                                     p->payload, p->payload_len);
-
                 if (json_output_ctx->flags & LOG_JSON_PAYLOAD_BASE64) {
-                    unsigned long len = sizeof(packet_buf) * 2;
+                    size_t len = p->payload_len * 2 + 1;
                     unsigned char encoded[len];
-                    Base64Encode(packet_buf, offset, encoded, &len);
+                    Base64Encode(p->payload, p->payload_len, encoded, &len);
```

Comment:
It's a bit strange - my compiler doesn't see anything suspicious.
Maybe we should return unsigned long here? For instance, LogTlsLogPem() uses unsigned long when calling Base64Encode.